index.js

```diff
@@ -44,6 +46,52 @@ module.exports = function (log, isReady) {
 
   var ready = Obv()
   ready.set(isReady !== undefined ? isReady : true)
+
+  var mapStream = opts => {
+    var getValue
+    var setValue
+    var hasNoValues = opts.values === false
+
+    if (opts.seqs) {
+      getValue = data => data.value
+      setValue = (data, value) => {
+        data.value = value
+        return data
+      }
+    } else {
+      getValue = data => data
+      setValue = (data, value) => value
+    }
+
+    return paramap((data, cb) => {
+      var err = null
+      if (hasNoValues)
+        return cb(err, data)
+
+      mapper(err, getValue(data), (_err, value) => {
+        cb(_err, setValue(data, value))
+      })
+    })
+  }
+
+  var streamPullSteps = opts => {
+    var steps = [ log.stream(opts), Looper ]
+    if (mapper)
+      steps.splice(1, 0, mapStream(opts))
```

Comment:
why return an array if it just gets passed to pull immediately anyway?
just have a createLogStream(opts) function that does returns the stream: `pull(log.stream(opts), opts.values !== false ? mapper : null, Looper)`

oh and seqs is enabled by default (see pull-cursor) so it should be `opts.seqs !== false` to check if it's enabled, not `opts.seqs`

Also, you are still creating the paramap stream even if you don't need it. check wether you need it before you create it.

I also find creating two function setValue and getValue to be confusing (because each section of code is not self contained, have to scan back to see where that function is defined, which is two places, and then figure out which one is happening), here is how I'd write it:
```
function Mapper(opts) {
  if(opts.values === false) //no values, can just pass through
    return
  else if(opts.seqs === false) //if seq has been explicitly disabled
    return Paramap(map)
  else //seqs is enabled
    return Paramap(function (data, cb){
      map(data.value, function (err, value) {
        cb(err, {seq: data.seq, value: value })
      })
    })
}
```
It's also less code, which I feel is a valuable guide regards clarity. Anyway, this aspect is obviously subjective, I'm just asking to see what your opinion is - I'd still merge this if you prefer to do it this way, as long as it skips creating an unused paramap.

Reply:
Sweet, I didn't realize I could just pass `null` and skip that step entirely, thanks! I've used that technique in both the `get()` and `stream()` methods to remove some boilerplate, and I've also streamlined that function to remove `(g|s)etValue()` and that unused paramap.

The only utility function I'm using now is that `mapWrap(cb)` function, which takes `mapper(err, data, cb)` and makes it accept `(err, data)` with an implicit callback. Please rename or reorganize anything in a way that makes more sense to you, my only preference is building a good solution that doesn't cause any future headaches. :upside_down_face: